az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -209,6 +200,7 @@ public static File getHadoopTokens(HadoopSecurityManager hadoopSecurityManager,
       throw new HadoopSecurityManagerException("Failed to create the token file.", e);
     }
 
+    log.info("prefetching hadoop token...");
```

Comment:
It's a fine idea. I just wonder if this is the best spot to log the message. If another client calls `prefetchToken` from somewhere else, wouldn't it be nice if we logged that as well? I think you are trying to log *before* we enter the synchronized method, in case that blocks. That's understood.  Would it be too much trouble to make `prefetchToken` an unsynchronized wrapper around the (private) `do_prefetch...`? 
Sorry, I think this was meant to be a quick change -- please feel free to ignore.

Reply:
@li-ygerchikov thanks and updated. Discussed with @HappyRay offline , the code is over complicated to reason around, but one side benefit of being synchronized i can think of is to prevent other threads hitting the job history server/metastore when the server is down or busy, which otherwise would make server even busier. Although it might not be the original motivation of being synchronized here.